util/coding.h

```diff
@@ -32,6 +32,7 @@ namespace rocksdb {
 const unsigned int kMaxVarint64Length = 10;
 
 // Standard Put... routines append to a string
+extern void PutFixed16(std::string* dst, uint16_t value);
```

Comment:
The changes here in coding.h might be a good candidate to split into a separate PR, with their own unit test in coding_test.cc, instead of clubbing them with DataBlockHashIndex discussion and implementation.